### PR TITLE
Added sql command feature

### PR DESCRIPTION
**Pull request by JamesRichards07**
Added sql commands and moved current sql scripts/preferred statements into Ravenfire directory. I tried moving a couple things into the traits but it was more complicated than writing it in the sql file itself if that makes sense. Let me know if you see anything which could/you want moved into sql traits.

### CODE REVIEW

**Comment by JamesRichards07** [comment]
Thought about adding an example sql script so users have a template to work off of.

File: src/Application/SqlScripts/CanHandleSql.php [hunk 1]
```diff
@@ -3,20 +3,37 @@
...
+
+    /**
+     * Limits the width of table columns to given length.
      *
      * @param $results
      * @param $db_columns
      * @return array
      */
-    public function handleResults($results, &$db_columns)
+    public function useAllColumnsHandler($results, &$db_columns, int $columnLength)
```
Comment (JamesRichards07) [inline src/Application/SqlScripts/CanHandleSql.php]: Can set default $columnLength to 12.

File: src/Magpie.php [hunk 2]
```diff
@@ -91,10 +88,22 @@ public function run()
...
+            /** @var AbstractSource $source_class */
+            foreach ($source_class::getNewCommands() as $command_class) {
+                if (class_exists($command_class)) {
+                    $application->add(new $command_class($this));
+                }
+            }
+        }
+
+        /** @var AbstractPrimaryEntity $primary_entity_class */
+        foreach ($this->getPrimaryEntity()::getNewCommands() as $command_class) {
```
Comment (JamesRichards07) [inline src/Magpie.php:102]: Example of when not to use a static method.

File: src/Ravenfire/BoardGameGeek/BoardGameGeekSource.php [hunk 3]
```diff
@@ -12,6 +15,15 @@
...
 {
+//    public static function getNewCommands(): array
+//    {
+//        return [
+//            SqlCountScript::class,
+//            SqlJoinScript::class,
+//            SqlFindScript::class
+//        ];
+//    }
+
```
Comment (JamesRichards07) [inline src/Ravenfire/BoardGameGeek/BoardGameGeekSource.php]: Please remove.

File: src/Ravenfire/Sql/SqlCountScript.php [hunk 4]
```diff
@@ -49,19 +49,18 @@ public function execute(InputInterface $input, OutputInterface $output)
 
         $table = $input->getArgument('table');
         $column = $input->getArgument('column');
-        $column_name = $input->getArgument('columnName');
+        $column_name = $input->getOption('columnName');
+        if ($column_name === null) $column_name = $column;
```
Comment (JamesRichards07) [inline src/Ravenfire/Sql/SqlCountScript.php]: Wrap in brackets.

File: src/Ravenfire/Sql/SqlCountScript.php [hunk 5]
```diff
@@ -76,14 +75,17 @@ public function execute(InputInterface $input, OutputInterface $output)
      * @param $column_name
      * @return mixed
      */
-    public function index($table, $column, $column_name)
+    public function index($table, $column, $column_name = null, $priority = null)
     {
         if ($this->checkTableAndColumnExist($table, $column)) {
```
Comment (JamesRichards07) [inline src/Ravenfire/Sql/SqlCountScript.php]: Add commet.

File: src/Ravenfire/Sql/SqlFindScript.php [hunk 6]
```diff
@@ -73,13 +75,13 @@ public function execute(InputInterface $input, OutputInterface $output)
      * @param $value
      * @return mixed
      */
-    public function index($table, $column, $value)
+    public function index($table, $column, $value, $operator)
     {
         if ($this->checkTableAndColumnExist($table, $column)) {
```
Comment (JamesRichards07) [inline src/Ravenfire/Sql/SqlFindScript.php]: Check for valid operator

File: src/Ravenfire/Sql/SqlJoinScript.php [hunk 7]
```diff
@@ -55,12 +55,9 @@ public function execute(InputInterface $input, OutputInterface $output)
 
         $db_columns = [];
 
-        $rows = $this->handleResults($results, $db_columns);
+        $rows = $this->useAllColumnsHandler($results, $db_columns, 12);
```
Comment (JamesRichards07) [inline src/Ravenfire/Sql/SqlJoinScript.php]: Remove 12 as a it will be set as default.

**Review by electricjones — APPROVED** [review]

**Pull request merged**